src/native/libs/System.Globalization.Native/pal_casing.m

```diff
@@ -0,0 +1,112 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_locale_internal.h"
```

Comment:
Why do we need this header?

Reply:
This is needed as `pal_casing.m` and `pal_casing.c` are using same header file `pal_casing.h`.
In `pal_casing.h `there are ICU related types (for example `UChar` ), if this header is not included it complains like `error: unknown type name 'UChar'`

Reply:
Included instead `pal_icushim_internal.h`